specs/core/0_beacon-chain.md

```diff
@@ -1604,33 +1618,31 @@ Below are the processing steps that happen at every slot.
 
 #### Block roots
 
-* Let `previous_block_root` be the `hash_tree_root` of the previous beacon block processed in the chain.
-* Set `state.latest_block_roots[(state.slot - 1) % LATEST_BLOCK_ROOTS_LENGTH] = previous_block_root`.
+* Set `state.latest_block_roots[state.slot % LATEST_BLOCK_ROOTS_LENGTH] = get_block_root(state, state.slot - 1)`.
 * If `state.slot % LATEST_BLOCK_ROOTS_LENGTH == 0` append `merkle_root(state.latest_block_roots)` to `state.batched_block_roots`.
 
 ### Per-block processing
 
 Below are the processing steps that happen at every `block`.
 
-#### Slot
-
-* Verify that `block.slot == state.slot`.
-
-#### Block signature
+#### Block header
 
+* Verify that `block.header.slot == state.slot`.
+* Verify that `block.header.parent_root == get_block_root(state, state.slot - 1)`.
+* Verify that `block.header.block_root == hash_tree_root(block.body)`.
+* Set `state.latest_block_roots[state.slot % LATEST_BLOCK_ROOTS_LENGTH] = block.header.block_root`.
```

Comment:
One problem I see with this approach is that if only block bodies are put into the block roots accumulator and state roots are not, then we lose the implicit accumulator of state roots. I would say the ability to access historical state roots through the accumulator is extremely important; along with many other arguments, it's where receipts will ultimately live.

Reply:
Good catch! Implemented `historical_batchings` as discussed on the call today. cc @djrtwo